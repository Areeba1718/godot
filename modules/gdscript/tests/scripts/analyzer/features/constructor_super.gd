func test():
	var _sub1 = SubClass1.new("test")
	var _sub2 = SubClass2.new()
	var _sub3 = SubClass3.new()

# Both with params.
class SuperClass1:
	func _init(param):
		prints("SuperClass1 init", param)

class SubClass1 extends SuperClass1:
	func _init(param):
		prints("SubClass1 init", param)
		super(param)

# Only super with params.
class SuperClass2:
	func _init(param):
		prints("SuperClass2 init", param)

class SubClass2 extends SuperClass2:
	func _init():
		prints("SubClass2 init")
		super("super")

# Super without params.
class SuperClass3:
	func _init():
		prints("SuperClass3 init")

class SubClass3 extends SuperClass3:
	func _init():
		prints("SubClass3 init")
		super()
