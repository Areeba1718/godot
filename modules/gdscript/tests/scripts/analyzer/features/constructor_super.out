GDTEST_OK
>> WARNING
>> Line: 14
>> UNSAFE_CALL_ARGUMENT
>> The argument 1 of the function "_init()" requires the subtype "Variant" but the supertype "Variant" was provided.
SubClass1 init test
SuperClass1 init test
SubClass2 init
SuperClass2 init super
SubClass3 init
SuperClass3 init
